addon/devtoolsOpener.js

```diff
@@ -1,25 +1,36 @@
 let backgroundPort = browser.runtime.connect();
 
-async function createPanel() {
-    const extensionPanel = await browser.devtools.panels.create(
+function createPanel() {
+    browser.devtools.panels.create(
```

Comment:
it's largely a style question; since i ended up rewriting the bulk of this code i went with the style i prefer.
i prefer it because it avoids the need to sprinkle functions with `async` (aka the 'async virus').